datadog_checks_base/tests/base/utils/db/test_timed_cache.py

```diff
@@ -0,0 +1,47 @@
+# (C) Datadog, Inc. 2024-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+import pytest
+import time
+
+from datadog_checks.base.utils.db.timed_cache import TimedCache
+
+
+class TestTimedCache:
+    def test_set_and_get_item(self):
+        cache = TimedCache(600)  # Using a longer interval for most tests to avoid automatic clearing during testing
+        cache['test_key'] = 'test_value'
+        assert cache['test_key'] == 'test_value', "The value retrieved should match the value set."
+
+    def test_get_with_default(self):
+        cache = TimedCache(600)
+        assert (
+            cache.get('nonexistent', 'default') == 'default'
+        ), "Should return the default value when the key is not found."
+
+    def test_item_deletion(self):
+        cache = TimedCache(600)
+        cache['test_key'] = 'test_value'
+        del cache['test_key']
+        with pytest.raises(KeyError):
+            _ = cache['test_key']  # Attempting to access deleted key should raise KeyError
+
+    def test_expire(self):
+        cache = TimedCache(1)  # Set a short interval for testing auto-clearing
+        cache['test_key'] = 'test_value'
+        assert cache, "The cache should not be empty immediately after setting an item."
+        time.sleep(2)  # Wait enough time for the cache to clear itself
```

Comment:
We have usages of `freeze_time` around the codebase: https://github.com/DataDog/integrations-core/blob/master/datadog_checks_downloader/tests/test_downloader.py#L138.

That might make this more reliable.

Reply:
freeze_time needs new freezegun dependency add to the base package. i feel it overkill to include a new dep